sponsorship/src/main/java/com/kyriba/conference/sponsorship/api/SponsorController.java

```diff
@@ -34,18 +37,26 @@
 
   @SuppressWarnings("unused")
   @ApiOperation(value = "Register the sponsor")
-  @PostMapping
   @ApiResponses(value = {
-      @ApiResponse(code = 200, message = "OK", response = SponsorRegistrationResponse.class),
-      @ApiResponse(code = 401, message = "Failed to register the sponsor", response = SponsorRegistrationResponse.class)
+      @ApiResponse(code = 200, message = "OK"),
+      @ApiResponse(code = 401, message = "Failed to register the sponsor")
   })
+  @PostMapping
   SponsorRegistrationResponse register(@Valid @RequestBody SponsorRegistrationRequest request)
   {
-    final String randomId = "123";
-    final Sponsor sponsor = Sponsor.builder().id(randomId)
-        .name(request.getName())
-        .email(request.getEmail())
-        .build();
-    return new SponsorRegistrationResponse(sponsor.getId());
+    return new SponsorRegistrationResponse(sponsorService.createSponsor(request.getName(), request.getEmail()));
+  }
+
+
+  @SuppressWarnings("unused")

```

Comment:
Why do you need this annotation on all methods in controller?

Reply:
It is needed to suppress Idea inspection "Unused declaration". Would it be better to move it to class level?

Reply:
Just ignore this warning in IDEA for controllers.